### ISSUE TITLE
Spark7111/7162 (meta-fulan) open issues

### ISSUE THREAD

**Issue by MastaG** [PLi-metas/pli-extras#39]
Instead of opening an issue everytime something goes wrong I think it's best to have just one issue for all spark related problems.

For the past weeks I've been working on the various problems reported by @lprot and others.

These are the fixes / changes so far.

- Multimedia
meta-fulan doesn't use PLi's servicemp3, instead it uses servicemp3epl from here:
https://github.com/PLi-metas/servicemp3epl
It contains two playback services:
servicemp3 - Modified by Taapat for spark receivers
servicelibp - LibePlayer3 playback backend

The recipe for these services can be found here:
https://github.com/PLi-metas/meta-fulan/blob/develop/recipes-openpli/enigma2/enigma2-plugin-systemplugins-servicemp3epl.bb

With the PACKAGECONFIG variable you can decide which ones to enable.
However, if you decide to include the libeplayer3 backend, you cannot use it together with ServiceApp.
That's why by default only gstreamer (servicemp3) is enabled.

Both libeplayer3 and exteplayer3 use ffmpeg for playback, which works better compared to gstreamer for HLS streams (because the slow CPU of these receivers.)

Also because ServiceApp + exteplayer3 has improved a lot for playing back HLS streams, I've decided to disable libeplayer3 and ship ServiceApp + exteplayer3 by default (also enabled by default).

Also I've merged the latest fixes for servicemp3.cpp as well.
See: https://github.com/PLi-metas/servicemp3epl/commits/master/servicemp3/servicemp3.cpp
So you can also disable ServiceApp and play your media with the dvbmediasink.

To sum things up, both servicemp3 (gstreamer dvbmediasink) and serviceapp (exteplayer3) now work fine.

- Other fixes
* HDMI-CEC driver is now loaded by default at the correct time
* zram is now always enabled
* both openpli-enigma2-image and pli-extras-enigma2 targets have been fixed
* the sh4 osd-position plugin now works and replaces the one from PLi
* jffs2 image is now created properly without errors
* JPEG acceleration has been disabled since it causes problems on some firmwares
* the openwebif should now display the correct remote for both spark and spark7162
* gstreamer playback does not stutter anymore (at least not for youtube videos)

Before reporting more issues / problems please make sure that you start clean.
So a clean checkout of openpli-oe-core and pli-extras WITHOUT ANY MODIFICATIONS.
Both spark and spark7162 targets should build fine now, including the feeds.

Things that still need to be figured out:
- Check whether the bootlogo displays (I don't have a TV atm, so I cant test)
- Check whether the correct firmware (audio.elf / video.elf) are being used for both spark and spark7162
- Other stuff..

**Comment by lprot** [comment]
I got following problems on this image:
1. Bootlogo doesn't show. If you replace it with openpli homebuild one it's ok.
2. HDMI-CEC plugin doesn't show up in the menu.

**Comment by lprot** [comment]
3. Gstreamer still stuttering.

Image was build with MACHINE=spark bitbake pli-extras-enigma2-image

**Comment by MastaG** [comment]
building a new image from scratch, after pli-extras/update.sh should:
- Install PLi's osd position instead of SH4 version
- Enable hdmi-cec plugin in AV settings menu
- Fix the OSD clock with a dot
- Only installs VFD plugin on 7162

**Comment by MastaG** [comment]
Update
- Switched dvb-moduels to Duckbox's developers
- Added skin_text_spark.xml (needs testing)

**Comment by MastaG** [comment]
Update
- Bump to latest duckbox git rev, this reverts the new videomode names used by OpenATV. Now 60Hz is working again.

**Comment by kueken** [comment]
I think the pti.ko still needs to be patched.

**Comment by lprot** [comment]
That was always so.

**Comment by kueken** [comment]
No, because before that the driver was based on the OpenArp / Taapat Branch.
There was the patch included.

**Comment by persianpros** [comment]
@kueken you can be a part of PLi-metas if you want and work on fulan 👍

**Comment by kueken** [comment]
Thank you, but I suspect my knowledge is too small ;-)

**Comment by persianpros** [comment]
That's not true at all, nobody is perfect, I know something you don't know and you know something I don't know, it's how nature works so helping each other is what makes sense.

I can't offer much except my friendship so would you please accept?

**Comment by MastaG** [comment]
That's true my friend.
And @kueken I've seen your work across multiple projects for sh4 :)
We'd be honored to have you contributing your fixes and enhancements.
I'm no hardcore C++/python programmer either.. but we're doing our best to bring the PLi experience to as many hardware as possible.
But it's a hobby after all, so only if you have the time :)

Thanks for the patch btw, will apply it in the next couple of days :+1:

**Comment by kueken** [comment]
I noticed something else
Fulan apps do not fit the structure of the driver.
libmmeimage / libmme-host

It is probably better to dodge apps on the ddt.
As in the OE-git was also made by skl.
In the e2 that would also have to be adjusted in picload.cpp.
On the other hand, it can also be completely deactivated and tested.
I can not find any disadvantages.

**Comment by lprot** [comment]
fp_control doesn't match aotom.ko or support of fp_control in aotom.ko is broken on both 7111 and 7162. That was so before switching to duckbox driver too.
Filed the issue here too https://github.com/Duckbox-Developers/driver/issues/2

**Comment by kueken** [comment]
This is not a driver problem, but rather a plugin problem.
You can address the LED directly

**Comment by lprot** [comment]
That is because if you replace aotom.ko in pliextras fw with aotom.ko from taapat openpli 5 image, fp_control works correctly. aotom.ko got some regression :(
And fp_control is not a plugins it's elf binary.

**Comment by kueken** [comment]
You have read my commit about it?
Apps and drivers do not match.

Try Build with Apps from ddt

**Comment by persianpros** [comment]
@MastaG @kueken @lprot @Taapat

I did cancel MastaG's invites for you as they were only for fulan related repos and I sent new invitations again so you could access all PLi-metas :)

You can public your membership here: https://github.com/orgs/PLi-metas/people

It's an honor to have you as our friends 👍 

Sorry for any inconvenience.

**Comment by lprot** [comment]
Thx, got a sample config for 7111/7162? And can you send Mastag you your phone number so we can talk on whatsup/viber?

**Comment by kueken** [comment]
it comes from
https://github.com/PLi-metas/enigma2-openpli-fulan/blob/master/lib/driver/vfd.cpp

It makes little sense sense in current images both ways to drive.
Use of old vfd.cpp / h and
https://github.com/PLi-metas/meta-fulan/blob/develop/recipes-bsp/drivers/fulan-dvb-modules/aotom_spark_procfs.patch

But much is possible.
But it is not a mistake of the driver, but just another way.
Directly the fp_control is no longer needed.
Indirectly, things are redirected internally.

Code for vfdstandby.cfg sample in /etc

`#Shutdown display ?
DISPLAY=TRUE
#Custom output, only available if in standby
#without using parameter -a and -d
DISPLAYCUSTOM=    %H:%M:%S    
#enabling network in standby?
NFSBOOT=TRUE`

You can get a solution from the driver / apps and appropriate patching via [Audionik ](https://github.com/Audioniek).
I find a lot of work for a little LED stuff.

**Comment by MastaG** [comment]
I think the two major issues which are still open are:
- Tuning to dvb-t2 plp channels.. this works in OpenATV currently uses the exact same driver, but not in our enigma2-openpli-fulan. I've been comparing various files (dvb.cpp frontend.cpp servicedvb.cpp), but it's a bit beyond my league I guess..

- VFD problems
I still need to look into this.. but I've reverted all my tryfixes and workarounds for the time being.

**Comment by kueken** [comment]
try this way
https://github.com/kueken/enigma2-openpli-fulan/commit/98fa1ecd04af71771667444655b4ea75a8aa8ca2

**Comment by MastaG** [comment]
Thanks a million @kueken 

Perhaps @lprot can spin a new image for testing :)

My 7162 only features dvb-t (no t2) and we don't receive plp (I believe).

**Comment by lprot** [comment]
That fix didn't work. 7162 owners still report that scanning is working ok. But when you trying to open the service you get error "No signal or tuner is configured incorrectly"

**Comment by kueken** [comment]
The scan was possible before? I can not test it unfortunately. Will try everything again with customization via libc header 3.4.100 to create. VFD with old solution via direct use of fp_control has not yet worked out.

**Comment by lprot** [comment]
The scan is ok on both old and new drivers. On old drivers, DVB-T2 scan was finding nothing, but DVB-T scan was finding DVB-T2 channels onlu with PLP=0. And you had ability to zip to that channels ok. 
On the new driver you can scan DVB-T2 with PLP=0,1,2 ok. But if you try to zip to that channels you get the error.

**Comment by kueken** [comment]
Ok,
if I understand this correctly, then way in the right direction.
Be the things transferred and continue to look.

**Comment by lprot** [comment]
Every time the guys test, they install clean compiled image and scan channels, then try to zip to them.

**Comment by lprot** [comment]
On the latest image on 7162 DVB-T2 works like magic. 
But we have regression with bootlogo and green led on vfd. Bootlogo doesn't show anymore. Green LED gone. Still have the problem with fp_control - we need it to make red led in deep standby.

**Comment by kueken** [comment]
I think it's not just magic, but what's the point.
regarding LED behavior the assignment has to be modified in the pli-image.
Directly addressing the fp_control I had already explained.
It is not intended anymore.
I also have to build and test everything myself first, it will take a while.
A working image is more important to me than an LED or a boot logo.
In the construction environment something changes every now and then and then the structures have to be adapted again.
The old solution via VFD, I've tried for a while, but that will be so nothing and has only devoured a lot of time. Therefore discarded the old way via direct use fp_control again.

**Comment by persianpros** [comment]
As we're using our forks for driver and kernel, I did cleanup the patches:

https://github.com/PLi-metas/meta-fulan/tree/develop/recipes-bsp/drivers/fulan-dvb-modules

https://github.com/PLi-metas/meta-fulan/tree/develop/recipes-bsp/linux/linux-fulan

Now if you have time please test the remaining patches and if you found them useful we would apply them for all :)

As I said here: https://forums.openpli.org/topic/57419-howto-build-unofficial-pli-for-your-unsupported-receiver/?view=findpost&p=934755 now meta-fulan could compile the images but in the spark way but we need other methods to boot them via usb, any idea?

Test images are here: http://www.mediafire.com/folder/p0hyt9w5cq4d9/pli-extras

**Comment by kueken** [comment]
https://github.com/Audioniek/flash

Since most boxes are integrated.
USB flash was already discussed in the pli forum

**Comment by MastaG** [comment]
@kueken @persianpros @lprot 
I've tried a fresh build for my spark7162 yesterday and I've also received my DVB-T antenna last week,
I can confirm that both DVB-T (KPN Digitenne) and DVB-C (Ziggo) are working great!
Thanks a lot guys, for all the driver and kernel enhancements!

I still need to look into the bootlogo problem.
Will hope to find some time later this week.

**Comment by kueken** [comment]
The boot logo is relatively easy to manage locally.
aiograb from OE-git expects a bootlogo.mvi.
Pli takes the detour via 'switchoff.mvi'.
Switchoff does not work, so the black bootlogo stays.
If you swap locally in the OpenPli area the bootlogo.mvi with what I have tried meta fulan, then it works with the bootlogo

**Comment by persianpros** [comment]
We can add a symlink ...

**Comment by MastaG** [comment]
@kueken 
I believe 'showiframe' is the package used to render the bootlogo (which is actually a mpeg4 still i-frame) into the video-decoder.

The openpli-bootlogo package installs the logo in /usr/share/bootlogo.mvi.
/usr/share/backdrop.mvi and /usr/share/bootlogo_wait.mvi are just symlinks to /usr/share/bootlogo.mvi but they are not called on init.

This is the init script:
root@spark7162:/usr/share# cat /etc/init.d/bootlogo 
if [ "$1x" == "stopx" ]
then
        exit 0
fi

# avoid the console messages clobbering our logo
[ -f /sys/class/vtconsole/vtcon1/bind ] && echo 0 > /sys/class/vtconsole/vtcon1/bind
# and set the correct videomode before showing the bootlogo
[ -f /etc/videomode ]  && [ -f /proc/stb/video/videomode ] && cat /etc/videomode > /proc/stb/video/videomode

BOOTLOGO=/usr/share/bootlogo.mvi
[ -f /etc/enigma2/bootlogo.mvi ] && BOOTLOGO=/etc/enigma2/bootlogo.mvi
/usr/bin/showiframe ${BOOTLOGO}

As you can see it will take /usr/share/bootlogo.mvi (which is not a symlink) or /etc/enigma2/bootlogo.mvi if it exists.

The default homebuild logo doesn't show for me, but other logo's do work.
Perhaps because the original logo is too small?
I'll have to check the source code of showiframe.c for this.

Anyways, I've renamed the openpli-bootlogo recipe to vision-bootlogo which will get installed when building the vision-enigma2-image target.
We still need to replace the bootlogo.mvi with our own though.

**Comment by MastaG** [comment]
Furthermore, I've updated the showiframe.c for sh4.
The new patch will convert OpenPLi's showiframe.c to OE-A's showiframe.c with OE-A's showiframe-sh4.patch applied.
See: https://github.com/PLi-metas/meta-fulan/commit/c1a1a32bdbe2fb929bb1e32725aef232a3e4c2ef
Perhaps all bootlogo issues are solved now.

**Comment by woooow1** [comment]
Hello guys, first of all, big thanks for hard work! Now on problems:

1. Im able to build image from latest meta for spark 7111
But i have bug for some sd channels in hellas sat. This problem is long time ago, but no one has manage to solve it. Tp is: 12.464 H  30000 2/3 DVB-S2/8PSK (some of channels with problems are FTA). When watch channels via stream from webif, there is no problem. But on tv, picture start to flickering and enigma2 crash! In attachment is crash log. Strange part of log for me is:

[eDVBScan] the current channel's ID was not corrected - not adding channel.
Backtrace:
/usr/bin/enigma2(_Z17handleFatalSignaliP9siginfo_tPv) [0x462CB8]
/usr/lib/libpython2.7.so.1.0(PyObject_SetAttr) [0x29D94DCC]
-------FATAL SIGNAL
and:

<4>failed to get output window -22
<4>failed to set output window -22

Any help is welcome!!!

2: After build image bootlogo, not work!

[enigma2_crash.log](https://github.com/PLi-metas/pli-extras/files/2430577/enigma2_crash.log)

**Comment by kueken** [comment]
Error because of the channel comes from the e2 system.
Then have all the boxes. Let's see if you find something there.

Bootlogo
We have not yet come to the actual error on the track.
Since you build yourself, swap in the area of ​​the pli folder, the boot logo (bootlogo.mvi) just locally with a to your liking.
Then it works after the construction.
